x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateMetricFieldMapper.java

```diff
@@ -0,0 +1,339 @@
+                        throw new IllegalArgumentException("Aggregate metric [" + metricName +
+                            "] does not exist in the mapping of field [" + fieldType.name() + "]");
+                    } else {
+                        context.addIgnoredField(fieldType.name());
+                        continue;
+                    }
+                }
+
+                // Make sure that the value is a number. Probably this will change when
+                // new aggregate metric types are added (histogram, cardinality etc)
+                if (e.getValue() instanceof Number == false) {
+                    if (ignoreMalformed.value() == false) {
+                        throw new IllegalArgumentException("Aggregate metric [" + metricName +
+                            "] of field [" + fieldType.name() + "] must be a number");
+                    } else {
+                        context.addIgnoredField(fieldType.name());
+                        continue;
+                    }
+                }
+
+                // Make sure that value_count is not a negative value
+                if (Metrics.value_count.toString().equals(metricName) == true) {
+                    if (e.getValue() instanceof Integer == false) {
+                        if (ignoreMalformed.value() == false) {
+                            throw new IllegalArgumentException("Aggregate metric [" + metricName +
+                                "] of field [" + fieldType.name() + "] must be an integer number");
+                        } else {
+                            context.addIgnoredField(fieldType.name());
+                            continue;
+                        }
+                    }
+
+                    Integer metricValue = (Integer) e.getValue();
+                    if (metricValue.intValue() < 0) {
+                        if (ignoreMalformed.value() == false) {
+                            throw new IllegalArgumentException("Aggregate metric [" + metricName +
+                                "] of field [" + fieldType.name() + "] must not be a negative number");
+                        } else {
+                            context.addIgnoredField(fieldType.name());
+                            continue;
+                        }
+                    }
+                }
+
+                parsedMetricsMap.put(metricName, e.getValue());
+            }
+        } else {
+            throw new MapperParsingException("Cannot parse aggregate metric for field [{}]",
+                null,fieldType.name());
+        }
+
+        // TODO: Implement null_value support?
+        if (parsedMetricsMap.keySet().containsAll(metrics.value()) == false
+            && ignoreMalformed.value() == false) {
+            throw new IllegalArgumentException("Aggregate metric field [" + fieldType.name() +
+                "] must contain all metrics " + metrics.value().toString());
+        }
+
+        if (fieldType().hasDocValues()) {
```

Comment:
I wonder if we should override `setHasDocValues()` on the MappedFieldType so that it throws an exception if the user tries to disable doc values?  Since we don't index anything and only store doc values, seems like we shouldn't let the user disable them?

Not sure, WDYT?